### PR TITLE
fixSampleRateForQuery

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
Fix https://github.com/Azure/azure-sdk-for-java/issues/36830

Issue:
Sample Rate is not being honored in query/CosmosPagedFlux

Fix:
Guarded tracing and diagnostics behind sample rate.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java [hunk 1]
```diff
@@ -38,9 +40,12 @@
  * @see FeedResponse
  */
 public final class CosmosPagedFlux<T> extends ContinuablePagedFlux<String, T, FeedResponse<T>> {
+    private final static Logger LOGGER = LoggerFactory.getLogger(CosmosPagedFlux.class);
```
Comment (jeet1995) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java:43]: Unused variable - might be a spot bug.

**Review by jeet1995 — APPROVED** [review]
LGTM - thanks.

**Review by kushagraThapar — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DiagnosticsProvider.java [hunk 2]
```diff
@@ -270,8 +269,9 @@ private <T> void endSpanCore(
 
         switch (signal.getType()) {
             case ON_COMPLETE:
+                end(cosmosCtx, statusCode, 0, actualItemCount, requestCharge, diagnostics,null, context, true);
```
Comment (jeet1995) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DiagnosticsProvider.java:272]: The CI is complaining of a fall-through case. Missing `break`.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DiagnosticsProvider.java [hunk 3]
```diff
@@ -748,10 +745,12 @@ private void end(
             diagnostics,
             throwable)) {
 
-            this.handleDiagnostics(context, cosmosCtx);
+            if (!isFluxCompleted || !cosmosCtx.getDiagnostics().isEmpty() ) {
```
[the remark this answers is not in the document]
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DiagnosticsProvider.java:749]: Fixed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java:43]: Fixed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DiagnosticsProvider.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/DiagnosticsProvider.java:272]: Fixed

**Pull request merged**